### PR TITLE
UF-4189: Corrected control to return rejection for empty list

### PR DESCRIPTION
**Pull request by josc04**
- Adjusted control to return rejection when case service returns empty list for issued parking permits

### CODE REVIEW

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/controls/ExpiringParkingPermitControl.java [hunk 1]
```diff
@@ -64,13 +65,13 @@ private boolean hasExpiringParkingPermit(ErrandDTO errand) {
 	}
 
 	private boolean allExpired(List<GetParkingPermitDTO> parkingPermits) {
-		return parkingPermits.stream()
+		return isNotEmpty(parkingPermits) && parkingPermits.stream()
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/controls/ExpiringParkingPermitControl.java:68]: Why not like this instead:

```
return Optional.ofNullable(parkingPermits).orElse(emptyList()).stream.
   .filter/map/whatever(...)
   .anyMatch()
```

Then you can skip the new dependency import.

Comment (josc04) [reply src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/controls/ExpiringParkingPermitControl.java:68]: As it was before:
```
parkingPermits.stream()
    .map(GetParkingPermitDTO::getArtefactPermitStatus)
    .allMatch(status -> equalsIgnoreCase(status, STATUS_EXPIRED));
```
returned true when list was empty (which is not what the check should return, as there is no parking permit that is expired. 

The suggested `ofNullable().orElse()` fixes the possibility of the list being null, but will still return true for empty list.

The control is to check that there is at least one member in list and that all members has expired. I couldn't think of a way to do it with just stream and filters.

Comment (josc04) [reply src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/controls/ExpiringParkingPermitControl.java]: Changed code and removed dependency

**Comment by josc04** [comment]
Acted on PR comment

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/controls/ExpiringParkingPermitControl.java (hunk 1 above)
Comment (foobar-1337) [reply src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/controls/ExpiringParkingPermitControl.java]: Ok, but this should work:

```
		return Optional.ofNullable(affecteds).orElse(Collections.emptyList()).stream()
			   .filter(status -> equalsIgnoreCase(status, STATUS_EXPIRED))
			   .findAny()
			   .isPresent();
```

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**